compiler/src/dotty/tools/dotc/fromtasty/TASTYRun.scala

```diff
@@ -22,9 +22,9 @@ class TASTYRun(comp: Compiler, ictx: Context) extends Run(comp, ictx) {
       file.extension match
         case "jar" =>
           JarArchive.open(Path(file.path), create = false).allFileNames()
-            .map(_.stripPrefix(File.separator)) // change paths from absolute to relative
-            .filter(e => Path.extension(e) == "tasty" && !fromTastyIgnoreList(e))
-            .map(e => e.stripSuffix(".tasty").replace(File.separator, "."))
+            .map(_.stripPrefix("/")) // change paths from absolute to relative
+            .filter(e => Path.extension(e) == "tasty" && !fromTastyIgnoreList(e.replace("/", File.separator)))
```

Comment:
seems ok, but perhaps `fromTastyIgnoreList` should accept binary names (with `.` separator) rather than paths?

Reply:
Maybe. Out of scope for this PR.